From 00da1683364e58c6430a4577123d01037f8faddc Mon Sep 17 00:00:00 2001
From: Florian Fainelli <florian@openwrt.org>
Date: Mon, 28 Jan 2013 20:06:26 +0100
Subject: [PATCH 08/11] MIPS: BCM63XX: introduce BCM63XX_EHCI configuration
 symbol

This configuration symbol can be used by CPUs supporting the on-chip
EHCI controller, and ensures that all relevant EHCI-related
configuration options are selected. So far BCM6328, BCM6358 and BCM6368
have an EHCI controller and do select this symbol. Update
drivers/usb/host/Kconfig with BCM63XX to update direct unmet
dependencies.

Signed-off-by: Florian Fainelli <florian@openwrt.org>
---
 arch/mips/bcm63xx/Kconfig |    9 +++++++++
 drivers/usb/host/Kconfig  |    5 +++--
 2 files changed, 12 insertions(+), 2 deletions(-)

--- a/arch/mips/bcm63xx/Kconfig
+++ b/arch/mips/bcm63xx/Kconfig
@@ -13,11 +13,18 @@ config BCM63XX_OHCI
 	select USB_OHCI_BIG_ENDIAN_DESC if USB_OHCI_HCD
 	select USB_OHCI_BIG_ENDIAN_MMIO if USB_OHCI_HCD
 
+config BCM63XX_EHCI
+	bool
+	select USB_ARCH_HAS_EHCI
+	select USB_EHCI_BIG_ENDIAN_DESC if USB_EHCI_HCD
+	select USB_EHCI_BIG_ENDIAN_MMIO if USB_EHCI_HCD
+
 config BCM63XX_CPU_6328
 	bool "support 6328 CPU"
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
 	select BCM63XX_OHCI
+	select BCM63XX_EHCI
 
 config BCM63XX_CPU_6338
 	bool "support 6338 CPU"
@@ -39,18 +46,21 @@ config BCM63XX_CPU_6358
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
 	select BCM63XX_OHCI
+	select BCM63XX_EHCI
 
 config BCM63XX_CPU_6362
 	bool "support 6362 CPU"
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
 	select BCM63XX_OHCI
+	select BCM63XX_EHCI
 
 config BCM63XX_CPU_6368
 	bool "support 6368 CPU"
 	select SYS_HAS_CPU_BMIPS4350
 	select HAVE_PCI
 	select BCM63XX_OHCI
+	select BCM63XX_EHCI
 endmenu
 
 source "arch/mips/bcm63xx/boards/Kconfig"
